## PANDAS

In [1]:
import pandas as pd
pd.options.display.float_format = '{:,.2f}'.format

## BegInv

In [2]:
BegInv0 = pd.read_csv("BegInv.csv")

In [3]:
BegInv1 = BegInv0.rename(columns = {'onHand':'BOYQty16',"InventoryId":"InvID","startDate":"StartDate"})

In [4]:
BegInv1['BOYInv$16'] = BegInv1['BOYQty16'] * BegInv1['Price']
BegInv1.head(2)

InvID  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  BOYQty16  Price   StartDate  BOYInv$16  
0  750mL         8  12.99  2016-01-01     103.92  
1  750mL         7  10.99  2016-01-01      76.93

## Deliverable 1a

In [5]:
BegInv_Store = BegInv1[['Store','BOYQty16','BOYInv$16']].groupby('Store').sum()
BegInv_Store.head(2)

BOYQty16  BOYInv$16
Store                     
1         49917 838,079.59
2         52925 840,452.20

In [6]:
BegInv_Store.sort_values(by='BOYInv$16',ascending=False).head(10)

BOYQty16    BOYInv$16
Store                       
34       153852 3,291,170.24
73       162551 3,142,497.36
67       158996 3,079,578.63
66       149314 2,973,033.90
76       140208 2,952,418.44
69       144255 2,946,726.65
38       114368 2,232,698.77
55       119641 2,001,263.66
50        94720 1,649,808.22
79        95330 1,503,149.48

## Deliverable 1b

In [7]:
BegInv_Brand = BegInv1[['Brand','BOYQty16','BOYInv$16']].groupby('Brand').sum()
BegInv_Brand.head(2)

BOYQty16  BOYInv$16
Brand                     
58          281   3,650.19
60          288   3,165.12

In [8]:
BegInv_Brand.sort_values(by='BOYInv$16',ascending=False).head(10)

BOYQty16  BOYInv$16
Brand                     
3545      14499 463,823.01
1233      12016 432,455.84
8068      15341 383,371.59
4261      15499 340,823.01
3858      13649 327,439.51
2753       4625 286,703.75
8082       9287 278,517.13
8680       7066 275,503.34
2589       6766 270,572.34
3876      14829 266,773.71

## EndInv

In [9]:
EndInv0 = pd.read_csv("EndInv.csv")

In [10]:
EndInv1 = EndInv0.rename(columns = {'onHand':'EOYQty16',"InventoryId":"InvID","endDate":"EndDate"})

In [11]:
EndInv1['EOYInv$16'] = EndInv1['EOYQty16'] * EndInv1['Price']
EndInv1.head(2)

InvID  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  EOYQty16  Price     EndDate  EOYInv$16  
0  750mL        11  12.99  2016-12-31     142.89  
1  750mL         7  36.99  2016-12-31     258.93

## Deliverable 2a

In [12]:
EndInv_Store = EndInv1[['Store','EOYQty16','EOYInv$16']].groupby('Store').sum()
EndInv_Store.head(2)

EOYQty16    EOYInv$16
Store                       
1         79827 1,206,845.93
2         56671   850,884.06

In [13]:
EndInv_Store.sort_values(by='EOYInv$16',ascending=False).head(10)

EOYQty16    EOYInv$16
Store                       
50       260717 4,887,260.68
73       164589 3,254,662.81
67       163765 3,076,114.82
34       145829 3,074,616.75
76       143866 2,975,945.18
69       150848 2,968,678.82
66       144579 2,860,504.99
74       166015 2,803,645.13
38       129397 2,463,906.85
55       125584 2,234,836.35

## Deliverable 2b

In [14]:
EndInv_Brand = EndInv1[['Brand','EOYQty16','EOYInv$16']].groupby('Brand').sum()
EndInv_Brand.head(2)

EOYQty16  EOYInv$16
Brand                     
58          385   5,001.15
60          146   1,604.54

In [15]:
EndInv_Brand.sort_values(by='EOYInv$16',ascending=False).head(10)

EOYQty16  EOYInv$16
Brand                     
1233      15047 526,494.53
3545      16770 502,932.30
2753       7849 470,861.51
8068      15608 366,631.92
3405      12268 355,649.32
4261      16769 351,981.31
2757      11603 336,370.97
2589       7922 300,956.78
1376      13180 276,648.20
2585      10487 272,557.13

## Deliverable 3

In [16]:
InvBalBegEnd = pd.merge (left = BegInv1[["InvID", "BOYQty16", "BOYInv$16"]], 
                         right = EndInv1[["InvID", "EOYQty16", "EOYInv$16"]], 
                         how = "outer", on = ["InvID"]).fillna(0)
InvBalBegEnd.head()

InvID  BOYQty16  BOYInv$16  EOYQty16  EOYInv$16
0  1_HARDERSFIELD_58      8.00     103.92     11.00     142.89
1  1_HARDERSFIELD_60      7.00      76.93      0.00       0.00
2  1_HARDERSFIELD_62      6.00     221.94      7.00     258.93
3  1_HARDERSFIELD_63      3.00     116.97      7.00     272.93
4  1_HARDERSFIELD_72      6.00     209.94      4.00     139.96

## Invoices

In [17]:
Invoices = pd.read_csv("Invoices.csv")
Invoices["InvoiceDate"] = pd.to_datetime(Invoices["InvoiceDate"])
Invoices["Year"] = Invoices["InvoiceDate"].dt.year
Invoices = Invoices[Invoices["Year"] == 2016]
Invoices.head(2)

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   

       PODate     PayDate  Quantity  Dollars  Freight Approval  Year  
0  2015-12-21  2016-02-16         6   214.26     3.47      NaN  2016  
1  2015-12-22  2016-02-21        15   140.55     8.57      NaN  2016

## Deliverable 4a

In [18]:
TenVendInv = Invoices[["VendorNumber", "VendorName", "Dollars"]].groupby('VendorNumber').sum()
TenVendInv.sort_values(by='Dollars',ascending=False).head(10)

VendorName       Dollars
VendorNumber                                                                 
3960          DIAGEO NORTH AMERICA INC   DIAGEO NORTH AMERIC... 49,733,333.66
4425          MARTIGNETTI COMPANIES MARTIGNETTI COMPANIESMAR... 26,851,015.81
12546         JIM BEAM BRANDS COMPANY    JIM BEAM BRANDS COM... 23,539,441.79
17035         PERNOD RICARD USA          PERNOD RICARD USA  ... 23,460,507.21
480           BACARDI USA INC            BACARDI USA INC    ... 16,985,053.64
1392          CONSTELLATION BRANDS INC   CONSTELLATION BRAND... 15,187,741.32
1128          BROWN-FORMAN CORP          BROWN-FORMAN CORP  ... 13,139,921.12
9165          ULTRA BEVERAGE COMPANY LLP ULTRA BEVERAGE COMP... 12,433,070.97
3252          E & J GALLO WINERY         E & J GALLO WINERY ... 11,858,245.15
9552          M S WALKER INC             M S WALKER INC     ... 10,294,481.73

## Deliverable 4b

In [19]:
TenVendFrt = Invoices[["VendorNumber", "VendorName", "Freight"]].groupby('VendorNumber').sum()
TenVendFrt.sort_values(by='Freight',ascending=False).head(10)

VendorName    Freight
VendorNumber                                                              
3960          DIAGEO NORTH AMERICA INC   DIAGEO NORTH AMERIC... 250,899.75
4425          MARTIGNETTI COMPANIES MARTIGNETTI COMPANIESMAR... 139,976.94
12546         JIM BEAM BRANDS COMPANY    JIM BEAM BRANDS COM... 120,230.57
17035         PERNOD RICARD USA          PERNOD RICARD USA  ... 120,130.51
480           BACARDI USA INC            BACARDI USA INC    ...  86,153.58
1392          CONSTELLATION BRANDS INC   CONSTELLATION BRAND...  77,791.20
1128          BROWN-FORMAN CORP          BROWN-FORMAN CORP  ...  66,576.22
9165          ULTRA BEVERAGE COMPANY LLP ULTRA BEVERAGE COMP...  63,888.38
3252          E & J GALLO WINERY         E & J GALLO WINERY ...  59,594.41
9552          M S WALKER INC             M S WALKER INC     ...  52,199.20

## Deliverable 4c

In [20]:
Invoices1 = Invoices[["VendorNumber", "VendorName", "Freight", "Dollars", "Quantity"]].groupby(['VendorNumber',"VendorName"]).sum()
Invoices1["Frt%"] = ((Invoices1["Freight"] / Invoices1["Dollars"]) * 100)
Invoices1["FrtUnit"] = Invoices1["Freight"] / Invoices1["Quantity"]

In [21]:
HighFrt = Invoices1
HighFrt = HighFrt[HighFrt["Dollars"] >= 250000]
HighFrt.sort_values(by='Frt%',ascending=False).head(10)

Freight       Dollars  Quantity  \
VendorNumber VendorName                                                       
9625         WESTERN SPIRITS BEVERAGE CO   1,746.69    323,948.73     52243   
1590         DIAGEO CHATEAU ESTATE WINES   7,083.48  1,331,574.55    183317   
9744         FREDERICK WILDMAN & SONS      3,885.16    736,944.55     68087   
17031        FLAG HILL WINERY & VINEYARD   1,477.16    281,173.13     19188   
516          BANFI PRODUCTS CORP           8,034.30  1,529,676.79    213967   
653          STATE WINE & SPIRITS          7,757.25  1,481,052.81    149590   
4425         MARTIGNETTI COMPANIES       139,767.62 26,810,799.70   2544805   
8892         TRINCHERO FAMILY ESTATES     11,225.72  2,175,075.75    340040   
8673         STE MICHELLE WINE ESTATES    15,507.50  3,007,382.16    411345   
9815         WINE GROUP INC               26,280.87  5,109,629.23    863638   

                                          Frt%  FrtUnit  
VendorNumber VendorName                                  
9625         WESTERN SPIRITS BEVERAGE CO  0.54     0.03  
1590         DIAGEO CHATEAU ESTATE WINES  0.53     0.04  
9744         FREDERICK WILDMAN & SONS     0.53     0.06  
17031        FLAG HILL WINERY & VINEYARD  0.53     0.08  
516          BANFI PRODUCTS CORP          0.53     0.04  
653          STATE WINE & SPIRITS         0.52     0.05  
4425         MARTIGNETTI COMPANIES        0.52     0.05  
8892         TRINCHERO FAMILY ESTATES     0.52     0.03  
8673         STE MICHELLE WINE ESTATES    0.52     0.04  
9815         WINE GROUP INC               0.51     0.03

In [22]:
LowFrt = Invoices1
LowFrt = LowFrt[LowFrt["Dollars"] >= 250000]
LowFrt.sort_values(by="Frt%").head(5)

Freight      Dollars  Quantity  \
VendorNumber VendorName                                                     
6359         OLE SMOKY DISTILLERY LLC     1,876.61   377,754.50    108698   
9819         TREASURY WINE ESTATES       14,470.47 2,903,971.85    487654   
3089         SIDNEY FRANK IMPORTING CO    8,258.10 1,656,429.63    179772   
3924         HEAVEN HILL DISTILLERIES    13,854.16 2,773,519.92    346500   
8352         LUXCO INC                    9,785.35 1,951,124.00    271443   

                                          Frt%  FrtUnit  
VendorNumber VendorName                                  
6359         OLE SMOKY DISTILLERY LLC     0.50     0.02  
9819         TREASURY WINE ESTATES        0.50     0.03  
3089         SIDNEY FRANK IMPORTING CO    0.50     0.05  
3924         HEAVEN HILL DISTILLERIES     0.50     0.04  
8352         LUXCO INC                    0.50     0.04

## Deliverable 4d

In [23]:
FrtCon = Invoices
FrtCon = FrtCon[FrtCon["Freight"] > 100]
FrtCon = FrtCon[FrtCon["Quantity"] <= 1000]
FrtCon = FrtCon[["VendorNumber", "VendorName", "Freight", "Dollars", "Quantity"]].groupby(['VendorNumber',"VendorName"]).sum()
FrtCon.sort_values(by="Freight",ascending=False).head()

Freight    Dollars  Quantity
VendorNumber VendorName                                               
2561         EDRINGTON AMERICAS          2,944.67 570,627.52     18120
8664         STOLI GROUP,(USA) LLC         349.02  14,299.98       975
7239         REMY COINTREAU USA INC        348.07   7,515.38       382
653          STATE WINE & SPIRITS          303.70   5,544.89       541
6785         PALM BAY INTERNATIONAL INC    242.73   5,533.18       715

## Deliverable 4e

In [24]:
FrtCon["Frt%"] = ((FrtCon["Freight"] / FrtCon["Dollars"]) * 100)
FrtCon["FrtUnit"] = FrtCon["Freight"] / FrtCon["Quantity"]
FrtCon.sort_values(by="Frt%",ascending=False).head(10)

Freight  Dollars  Quantity  Frt%  \
VendorNumber VendorName                                                      
653          STATE WINE & SPIRITS          303.70 5,544.89       541  5.48   
7239         REMY COINTREAU USA INC        348.07 7,515.38       382  4.63   
6785         PALM BAY INTERNATIONAL INC    242.73 5,533.18       715  4.39   
2555         DISARONNO INTERNATIONAL LLC   146.86 3,506.41       385  4.19   
6213         MHW LTD                       177.50 4,821.83       412  3.68   
4692         KOBRAND CORPORATION           185.93 5,336.48       427  3.48   
1485         CASTLE BRANDS CORP.           179.26 5,420.41       320  3.31   
8352         LUXCO INC                     196.61 6,139.47       819  3.20   
10000        MAJESTIC FINE WINES           218.05 7,234.88       875  3.01   
3089         SIDNEY FRANK IMPORTING CO     133.83 4,443.66       379  3.01   

                                          FrtUnit  
VendorNumber VendorName                            
653          STATE WINE & SPIRITS            0.56  
7239         REMY COINTREAU USA INC          0.91  
6785         PALM BAY INTERNATIONAL INC      0.34  
2555         DISARONNO INTERNATIONAL LLC     0.38  
6213         MHW LTD                         0.43  
4692         KOBRAND CORPORATION             0.44  
1485         CASTLE BRANDS CORP.             0.56  
8352         LUXCO INC                       0.24  
10000        MAJESTIC FINE WINES             0.25  
3089         SIDNEY FRANK IMPORTING CO       0.35

The average Freight % for the 10 highest Frt% vendors from 4c is 0.525, and the average Freight/Unit for the 10 highest Frt% vendors from 4c is 0.045.
The average Freight % for the 10 highest Frt% vendors from 4d is 3.838, and the average Freight/Unit for the 10 highest Frt% vendors from 4d is 0.446.
This shows that vendors with higher sales (above 250,000) generally have a much lower Freight % and Freight/Unit cost than vendors with lower sales (below 100), likely due to freight cost discounting that occurs as order sizes grow and shipments become more efficient.

## Deliverable 5

In [25]:
Sales = pd.read_csv("Sales.csv")
Sales["SalesDate"] = pd.to_datetime(Sales["SalesDate"])
Sales.sort_values(by = "SalesDollars", ascending = False).head(2)

InventoryId  Store  Brand            Description   Size  \
268144  33_HORNSEY_2589     33   2589  Jameson Irish Whiskey  1.75L   
277630   33_HORNSEY_800     33    800  Jameson Irish Whiskey  Liter   

        SalesQuantity  SalesDollars  SalesPrice  SalesDate   Volume  \
268144            686     26,061.14       37.99 2016-12-23 1,750.00   
277630            697     22,994.03       32.99 2016-12-14 1,000.00   

        Classification  ExciseTax  VendorNo         VendorName  
268144               1   1,260.52     17035  PERNOD RICARD USA  
277630               1     731.85     17035  PERNOD RICARD USA

In [26]:
Sales = pd.read_csv("Sales.csv")
Sales = Sales[Sales["Description"] == "Jameson Irish Whiskey"]
Sales = Sales[Sales["Size"] == "1.75L"]
Sales.head(2)

InventoryId  Store  Brand            Description   Size  \
4644  10_HORNSEY_2589     10   2589  Jameson Irish Whiskey  1.75L   
4645  10_HORNSEY_2589     10   2589  Jameson Irish Whiskey  1.75L   

      SalesQuantity  SalesDollars  SalesPrice   SalesDate   Volume  \
4644              3        113.97       37.99  2016-12-01 1,750.00   
4645              7        265.93       37.99  2016-12-03 1,750.00   

      Classification  ExciseTax  VendorNo         VendorName  
4644               1       5.51     17035  PERNOD RICARD USA  
4645               1      12.86     17035  PERNOD RICARD USA

In [27]:
Sales = pd.read_csv("Sales.csv")
Sales = Sales[Sales["Description"] == "Jameson Irish Whiskey"]
Sales = Sales[Sales["Size"] == "1.75L"]
TotalSales = Sales["SalesDollars"].sum()
print(TotalSales)

363678.27


In [28]:
Sales = pd.read_csv("Sales.csv")
TotalSales = Sales["SalesDollars"].sum()
print(TotalSales)

38279409.84000002


In [29]:
Sales = pd.read_csv("Sales.csv")
Sales = Sales[Sales["Description"] == "Jameson Irish Whiskey"]
Sales = Sales[Sales["Size"] == "1.75L"]
TotalQuantity = Sales["SalesQuantity"].sum()
print(TotalQuantity)

9573


In [30]:
Sales = pd.read_csv("Sales.csv")
TotalQuantity = Sales["SalesQuantity"].sum()
print(TotalQuantity)

2748321


In [31]:
Prices = pd.read_csv("Prices.csv")
Prices = Prices[["Description", "Size", "PurchasePrice"]]
Prices = Prices[Prices["Description"] == "Jameson Irish Whiskey"]
Prices = Prices[Prices["Size"] == "1750mL"]
Prices.head()

Description    Size  PurchasePrice
391  Jameson Irish Whiskey  1750mL          30.76

### Jameson Irish Whiskey (1.75L)

Sales Price: 37.99

Purchase Price: 30.76

Contribution Margin: 7.03 (18.50%)

Total December Sales: 363,678.27 (0.95%)

Total December Quantity: 9,573 (0.35%)

Total Profit: 67,298.19

The 1.75L Jameson Irish Whiskey sales made up only 0.35% of the count of units sold in December, but accounted for 0.95% of the total dollar sales for the month. The contribution margin for this product shows managers that it is creating profit at a rate relatively higher than other products offered. Management can use this knowledge to explore ways of increasing the product's contribution margin to generate profit at a higher rate, and/or promote this product to their consumers so increase overall sales/profit at the current rate.